src/pages/Register.tsx

```diff
@@ -1,38 +1,55 @@
-import {Column, Container, Input, Button} from '@froggy-app/lilypad';
-import {useState} from 'react';
-import {useDispatch, useSelector} from 'react-redux';
-import {useNavigate} from 'react-router-dom';
+import { Column, Container, Input, Button } from '@froggy-app/lilypad';
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 import selectAuthError from 'redux/selectors/selectAuthError';
 import registerAccount from 'redux/thunks/registerAccount';
 
 const passwordRules = [
   {
     label: 'An uppercase character',
-    valid: (value: string) => /^(?=.*[A-Z])/.test(value),
+    valid: (value: string) => /^(?=.*[A-Z])/.test(value)
   },
   {
     label: 'A lowercase character',
-    valid: (value: string) => /^(?=.*[a-z])/.test(value),
+    valid: (value: string) => /^(?=.*[a-z])/.test(value)
   },
   {
     label: 'A number',
-    valid: (value: string) => /^(?=.*[0-9])/.test(value),
+    valid: (value: string) => /^(?=.*[0-9])/.test(value)
   },
   {
     label: 'A special symbol (!@#$%^&*])',
-    valid: (value: string) => /^(?=.*[!@#$%^&*])/.test(value),
+    valid: (value: string) => /^(?=.*[!@#$%^&*])/.test(value)
   },
   {
     label: '12 to 64 characters',
-    valid: (value: string) => value.length >= 12 && value.length <= 64,
-  },
+    valid: (value: string) => value.length >= 12 && value.length <= 64
+  }
 ];
 
 const emailRules = [
   {
-    valid: (value: string) =>
-      /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(value),
-  },
+    valid: (value: string) => {
+      const tokens = value.split('@');
+
+      if (tokens.length != 2) {
+        return false;
+      }
+
+      // Section before the @ (as well as domain name) is inherently valid because of the inputs allowed chars
+      const domainStr = tokens[1];
+      const domainTokens = domainStr.split('.');
+
+      if (domainTokens.length != 2) {
+        return false;
+      }
+
+      const tld = domainTokens[1];
+
+      return /^[a-zA-Z]+$/.test(tld);
+    }
+  }
```

Comment:
Only notable change. Rest is prettier

Reply:
I believe this is still needed. The email input only allows certain characters, but it does not know once a full  and valid email has been entered. For example: 'abc' would be allowed, but we wouldn't consider it a valid email.

We want to know when a valid email is entered so that we can show the user feedback with the green checkmark.